### PR TITLE
Improve cron script

### PR DESCRIPTION
**Pull request by valentinperignon**
## Changes

- [x] Improved cron script that makes lose the current round for players who have not played after 24 hours

### CODE REVIEW

File: server/files-script/cron-tasks.js [hunk 1]
```diff
@@ -38,48 +38,54 @@ const removeDuelsTask = cron.schedule(
...
+        Object.keys(listOfDuels).map(async (duelID) => {
+          const currentDuel = listOfDuels[duelID];
 
-        let indexLastPlayed;
-        if (currentRound[0].time === null || currentRound[1].time === null) {
-          indexLastPlayed = currentRound[0].time ? 0 : 1;
-        } else {
-          indexLastPlayed = currentRound[0].time > currentRound[1].time ? 0 : 1;
-        }
+          let needToPlayID = 0;
```
Comment (FrancoisPog) [inline server/files-script/cron-tasks.js:64]: Is it the ID or the index of the player who needs to play ?

**Review by FrancoisPog — APPROVED** [review]
LGTM !

File: server/files-script/cron-tasks.js [hunk 2]
```diff
@@ -38,48 +38,54 @@ const removeDuelsTask = cron.schedule(
...
-        if (!duelsTime[data.du_id]) {
-          duelsTime[data.du_id] = [];
-        }
-        duelsTime[data.du_id].push({
-          user: data.us_login,
-          answers: data.re_answers,
-          time: lastTime === null ? null : new Date(lastTime),
+      const res = await queryPromise(sql);
+      res.forEach((element) => {
+        if (!listOfDuels[element.du_id]) listOfDuels[element.du_id] = [];
```
[the remark this answers is not in the document]
Comment (valentinperignon) [reply server/files-script/cron-tasks.js:51]: We need to ask @nathanaelhoun 
So let's ping @nathanaelhoun 
It's the first time I ping someone
And it's @nathanaelhoun

[the remark this answers is not in the document]
Comment (nathanaelhoun) [reply server/files-script/cron-tasks.js:51]: I don't like them either (no offense, poor one-line if), but it kinda makes sense here.

File: server/files-script/cron-tasks.js (hunk 1 above)
Comment (valentinperignon) [reply server/files-script/cron-tasks.js:64]: Yep! Do you want me to change the name of the variable?
I could rename it b

**Review by nathanaelhoun — APPROVED** [review]

File: server/files-script/cron-tasks.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (nathanaelhoun) [reply server/files-script/cron-tasks.js:51]: > We need to ask @nathanaelhoun
> So let's ping @nathanaelhoun
> It's the first time I ping someone
> And it's @nathanaelhoun

> Toujours des délais exorbitants...

File: server/files-script/cron-tasks.js (hunk 1 above)
Comment (nathanaelhoun) [reply server/files-script/cron-tasks.js:64]: I feel like "Var" would be more descriptive

**Pull request merged**